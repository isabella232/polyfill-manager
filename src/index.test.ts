import { Polyfill, Polyfills, applyPolyfills } from "."

function polyfillFactory(
  shouldBeApplied = true,
  name = "test polyfill",
  polyfill = jest.fn(() => Promise.resolve())
): Polyfill {
  return {
    name,
    test: () => !shouldBeApplied,
    polyfill
  }
}

test("logger", async () => {
  const testValues: Polyfills = [ [ polyfillFactory() ] ]

  const logger = jest.fn()
  await applyPolyfills(testValues, {
    debug: true,
    logger
  })
  expect(logger).toMatchSnapshot()
})

test("polyfill applied", async () => {
  const polyfill = polyfillFactory()
  const testValues: Polyfills = [ [ polyfill ] ]

  await applyPolyfills(testValues)
  expect(polyfill.polyfill).toHaveBeenCalledTimes(1)
})

test("polyfill execution order", async () => {
  const result = []

  // eslint-disable-next-line jest/valid-expect-in-promise
  const mockFacktory = (id) =>
    Promise.resolve().then(() => {
      result.push(id)
    })

  const polyfill1 = polyfillFactory(true, "test polyfill", jest.fn(() => mockFacktory(1)))
  const polyfill2 = polyfillFactory(true, "test polyfill", jest.fn(() => mockFacktory(2)))
  const polyfill3 = polyfillFactory(true, "test polyfill", jest.fn(() => mockFacktory(3)))

  const testValues: Polyfills = [ [ polyfill1 ], [ polyfill2 ], [ polyfill3 ] ]
  await applyPolyfills(testValues)
  expect(result).toMatchSnapshot()
})
