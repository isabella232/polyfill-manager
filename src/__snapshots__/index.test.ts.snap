// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Execute",
      "test polyfill",
    ],
    Array [
      "test polyfill",
      "successfully loaded",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`polyfill execution order 1`] = `
Array [
  1,
  2,
  3,
]
`;
